File: src/test/java/net/haesleinhuepf/clij/ops/examples/AutoThreshold.java

package net.haesleinhuepf.clij.ops.examples;

import java.io.IOException;

import org.junit.Test;

import net.haesleinhuepf.clij.ops.CLIJ_automaticThreshold.CLIJ_automaticThreshold;
import net.haesleinhuepf.clij.ops.CLIJ_close.CLIJ_close;
import net.haesleinhuepf.clij.ops.CLIJ_pull.CLIJ_pull;
import net.haesleinhuepf.clij.ops.CLIJ_push.CLIJ_push;
import net.imagej.ImageJ;
import net.imglib2.img.Img;

public class AutoThreshold {

	@Test
	public void run() throws IOException {
		ImageJ ij = new ImageJ();
		Img blobs = (Img) ij.io().open("https://bds.mpi-cbg.de/samples/blobs.png");
		ij.ui().show("input", blobs);
		// push image to GPU
		Object blobsGPU = ij.op().run(CLIJ_push.class, blobs);
		// run threshold
		Object maskGPU = ij.op().run(CLIJ_automaticThreshold.class, blobsGPU,
			"Otsu");
		// show result
		Object result = ij.op().run(CLIJ_pull.class, maskGPU);
		ij.ui().show("auto threshold", result);
		// cleanup
		ij.op().run(CLIJ_close.class, blobsGPU);
		ij.op().run(CLIJ_close.class, maskGPU);
		ij.op().run(CLIJ_close.class);

	}

	public static void main(String... args) throws IOException {
		AutoThreshold task = new AutoThreshold();
		task.run();
	}
}

File: src/test/resources/jython/autoThreshold.py
# CLIJ example ImageJ Ops jython: thresholding.py
#
# This script shows how to apply an automatic threshold method to an image in the GPU.
#
# Author: Robert Haase, rhaase@mpi-cbg.de
# Author: Deborah Schmidt, frauzufall@mpi-cbg.de
# May 2019
# ---------------------------------------------

#@ IOService io
#@ UIService ui
#@ OpService ops

blobs = io.open("https://bds.mpi-cbg.de/samples/blobs.png")
ui.show("input", blobs)
# push image to GPU
blobsGPU = ops.run("CLIJ_push", blobs)
# run threshold
maskGPU = ops.run("CLIJ_automaticThreshold", blobsGPU, "Otsu")
# show result
result = ops.run("CLIJ_pull", maskGPU)
ui.show("auto threshold", result)

#cleanup
ops.run("CLIJ_close", blobsGPU)
ops.run("CLIJ_close", maskGPU)
ops.run("CLIJ_close")